include/oneapi/dpl/pstl/hetero/utils_hetero.h

```diff
@@ -0,0 +1,152 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+// This file contains SYCL specific macros and abstractions
+// to support different versions of SYCL and to simplify its interfaces
+//
+// Include this header instead of sycl.hpp throughout the project
+
+#ifndef _ONEDPL_utils_hetero_H
+#define _ONEDPL_utils_hetero_H
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __internal
+{
+
+template <typename _Pred>
+struct equal_predicate
```

Comment:
Do the `equal_predicate` and `adjacent_find_fn` names need to be uglified?  I.e., `equal_predicate` replaced with `__equal_predicate`?

Reply:
That stuff was been just moved to another file.
Probably, to fix qualification issue everywhere does it make sense to prepare a separate patch?

Reply:
Ok. I see the namespace of these structs wasn't even changed by the move.  I agree that fixing the uglification issue everywhere would be best done as a separate patch then.